pkgs/applications/video/popcorntime/default.nix

```diff
@@ -0,0 +1,81 @@
+{ pkgs
+, writeShellScriptBin
+, makeWrapper
+, nodePackages
+, nwjs
+, fetchFromGitHub
+, xdg-utils
+, lib
+, fetchurl }:
+
+let
+
+  pname = "popcorntime";
+  version = "0.4.7";
+  name = "${pname}-${version}";
+
+  src = fetchFromGitHub {
+    owner = "popcorn-official";
+    repo = "popcorn-desktop";
+    rev = "v${version}";
+    sha256 = "sha256-Iz6wJPSQ5xcrPN43OTNOvpUS8sKCqKhx2FbGpLvjymU=";
+  };
+
+  bundle = "$out/usr/share/${pname}";
+
+  # We are using the Nwjs version referenced by popcorntime (0.44.5). Newer
+  # versions seem to fail.
+  # See:https://github.com/popcorn-official/popcorn-desktop/issues/2062
+  expectedNwjsVersion = "0.44.5";
+
+  nwjs' = nwjs.overrideAttrs (oldAttrs: rec {
+    version = expectedNwjsVersion;
+    src = fetchurl {
+      url = "https://dl.nwjs.io/v${version}/nwjs-v${version}-linux-x64.tar.gz";
+      sha256 = "sha256-LUs6wfQkpW8zH6WQOTcxSVOFGbQGtNg9EuiYGCsI3SQ=";
+    };
+  });
+
+
+in (import ./node.nix {
```

Comment:
did you try mkYarnPackage, it has a fetchyarndeps that works with FOD. It would prevent us from having this huge node-packages file.
everytime we commit more than 10000 lines, I feel bad for our future selves that will have to pay the consequences.
(if you are looking for an example, I think there is one in `hedgedoc`

Reply:
I‘ll have a look, tried it already but it didn‘t work https://github.com/NixOS/nixpkgs/pull/168121#issuecomment-1095283261

Reply:
I don't really like mkYarnPackage as it's not super intuitive.
You most probably have to override the buildPhase and the postConfigurePhase.
in hedgedoc they do
```
  postConfigure = ''
    rm deps/HedgeDoc/node_modules
    cp -R "$node_modules" deps/HedgeDoc
    chmod -R u+w deps/HedgeDoc
  '';

  buildPhase = ''
    runHook preBuild

    cd deps/HedgeDoc

    pushd node_modules/sqlite3
    export CPPFLAGS="-I${nodejs}/include/node"
    npm run install --build-from-source --nodedir=${nodejs}/include/node
    popd

    pushd node_modules/esbuild
    rm bin/esbuild
    ln -s ${lib.getBin esbuild-hedgedoc}/bin/esbuild bin/
    popd

    npm run build

    patchShebangs bin/*

    runHook postBuild
  '';
```

the way mkYarnPackage works is that it puts all the dependencies inside a `deps` directory, even the current package that you are building.
This is why they are cd'ing into the deps/hedgedoc

most likely you have to cd into `deps/popcorntime` and do a `yarn build` there.
You might have to cp the resulting dist directory into $out.
I know it's not super intuitive, but if you need to make mkYarnPackage work, you just need to `ls -lah` at some points to verify the directory structure and verify that all is needed.
In the case of hedgedoc they have additional dependencies that they needed to bring (esbuild and sqlite), hopefully you don't have to do the same dance.

of course, feel free to post any logs that look weird to you.